### ISSUE TITLE
BUG: Cannot build master branch of numpy in AIX.

### ISSUE THREAD

**Issue by KamathForAIX** [numpy/numpy#25244]
### Describe the issue:

Respected community members,

Hi,

Numpy is not building successfully with the current version of Vendor Meson in AIX. 

We also tried using the manual build option, but it looks like the features module is not available in the master branch of Meson.

The pip way fails with an unhandled python exception because it uses meson < 1.3.0

The current vender version or fork of Meson that Numpy uses is 1.2.99.

In AIX, meson versions < 1.3.0 had problems finding a.so file. This is because, by default, Meson archives a shared library in AIX, and due to the dl_open() mechanism of Python, it searches for a '.so' and not a '.a' archive file. This issue was fixed recently in Meson by AIX folks and the Meson community so that Python modules using the Meson build system in AIX can be built smoothly.

Therefore, I kindly request that the Numpy community update the vendor version of Meson to 1.3.0 or higher. Kindle let me know if any more information or help is needed so that we can fix this for Numpy and AIX

Have a nice day ahead.

Thanks and regards,

### Reproduce the code example:

```python
pip3.9 install numpy 
......
The Meson build system
  Version: 1.2.99
  Source dir: /numpy-1.26.2
  Build dir: /numpy-1.26.2/.mesonpy-hj2ou2ww/build
  Build type: native build
........
File "/numpy-1.26.2/vendored-meson/meson/mesonbuild/compilers/compilers.py", line 963, in get_command_to_archive_shlib
      return self.linker.get_command_to_archive_shlib()

ERROR: Unhandled python exception
  
      This is a Meson bug and should be reported!
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed
.........
```

### Runtime information:

When we manually build, using meson-1.3.0 we get

../meson_cpu/x86/meson.build:2:15: ERROR: Module "features" does not exist

  A full log can be found at /numpy/.mesonpy-f1ggpx8t/meson-logs/meson-log.txt

**Comment by rgommers** [comment]
Thank you for the report @KamathForAIX. We'll update our vendored Meson to the just-released 1.3.0, that will fix this problem. 

Glad to hear that AIX is working with Meson now.

**Comment by KamathForAIX** [comment]
Thanks @rgommers
